iceoryx_posh/experimental/include/iox/posh/experimental/node.hpp

```diff
@@ -62,10 +65,37 @@ class NodeBuilder
     IOX_BUILDER_PARAMETER(bool, shares_address_space_with_roudi, false)
 
   public:
+    /// @brief Determines which domain to use to register to a RouDi instance
+    /// @param[in] value to be used as domain ID
+    NodeBuilder&& domain_id(const uint16_t value) && noexcept;
```

Comment:
I changed it but I'm not sure if I like the API since one now has to use it like `domain_id(DomainId{42})` and I would not like it to be implicit convertible.

Reply:
But with this construct, you could add `explicit` in front of the ctor.

Reply:
That's already the case and therefore it is `domain_id(DomainId{42})` instead of just `domain_id(42)`